auth0/src/main/java/com/auth0/android/authentication/storage/SecureCredentialsManager.kt

```
@@ -111,86 +111,70 @@ public class SecureCredentialsManager @VisibleForTesting(otherwise = VisibleForT
+    override fun getSessionTransferCredentials(
+        parameters: Map<String, String>,
+        callback: Callback<SessionTransferCredentials, CredentialsManagerException>
+    ) {
         serialExecutor.execute {
-            lateinit var existingCredentials:Credentials
-            try{
+            lateinit var existingCredentials: Credentials
+            try {
                 existingCredentials = getExistingCredentials()
-            }catch (exception:CredentialsManagerException){
+            } catch (exception: CredentialsManagerException) {
                 callback.onFailure(exception)
                 return@execute
             }
             if (existingCredentials.refreshToken.isNullOrEmpty()) {
                 callback.onFailure(CredentialsManagerException.NO_REFRESH_TOKEN)
                 return@execute
             }
+
+            val request =
+                authenticationClient.fetchSessionTransferToken(existingCredentials.refreshToken!!)
             try {
-                val sessionCredentials =
-                    authenticationClient.fetchSessionToken(existingCredentials.refreshToken!!)
-                        .execute()
-                saveSsoCredentials(sessionCredentials)
+                if (parameters.isNotEmpty()) {
+                    request.addParameters(parameters)
+                }
+                val sessionCredentials = request.execute()
+                saveSessionTransferCredentials(sessionCredentials)
                 callback.onSuccess(sessionCredentials)
             } catch (error: AuthenticationException) {
                 val exception = when {
-                    error.isRefreshTokenDeleted ||
-                            error.isInvalidRefreshToken -> CredentialsManagerException.Code.RENEW_FAILED
+                    error.isRefreshTokenDeleted || error.isInvalidRefreshToken -> CredentialsManagerException.Code.RENEW_FAILED
```

Should we throw a distinct error here?